src/sql/src/plan/statement/ddl.rs

```diff
@@ -2483,11 +2491,23 @@ pub fn plan_create_sink(
         seen: _,
     } = with_options.try_into()?;
 
-    let cluster_config = source_sink_cluster_config(scx, "sink", in_cluster.as_ref(), size)?;
-
     // WITH SNAPSHOT defaults to true
     let with_snapshot = snapshot.unwrap_or(true);
 
+    let cluster_config = source_sink_cluster_config(scx, "sink", in_cluster.as_ref(), size)?;
+    // If using the default cluster, ensure we record which cluster that is.
```

Comment:
All of the statements that use this function get normalized after we insert the new cluster name into the `in_cluster` argument--i.e. it becomes immutable.

Looking at this again, it seems like that fixup should occur in the function itself rather than relying on the caller to do it.